### ISSUE TITLE
Add option to ignore no-trailing-spaces on lines without code

### ISSUE THREAD

**Issue by andrewvaughan** [eslint/eslint#2303]
Many IDEs provide formatting assistance by maintaining the indentation on lines without code.  For example:

```javascript
function myFunction() {
    var thisVar;
    
    // 4 spaces above this line
}
```

This will throw a `no-trailing-spaces` error.  An option should be added to the `no-trailing-spaces` rule that allows the engine to ignore the `no-trailing-spaces` rule on lines consisting entirely of whitespace characters.

Developers concerned with optimization should still remove blank-line whitespace, so it is recommended that the new option be `false` by default.

**Comment by nzakas** [comment]
Should this just be the default behavior?

**Comment by andrewvaughan** [comment]
I would say yes for server side (Node, etc) environments, but not for browser.

**Comment by nzakas** [comment]
What's the difference? If you're minifying code, extra whitespace certainly doesn't matter. Certainly you're using the same editor for both?

**Comment by andrewvaughan** [comment]
True - I'm for making it default.

**Comment by nzakas** [comment]
@btmills @lo1tuma thoughts on this change?

**Comment by btmills** [comment]
This seems useful. I'm not convinced it should be the default behavior, however, as that seems non-nobvious (I'm imagining an issue titled "no-trailing-spaces skips blank lines"). In such cases I prefer to have opt-in features rather than opt-out.

**Comment by nzakas** [comment]
Good point. So maybe `skipBlankLines` as an option?

**Comment by btmills** [comment]
Makes sense to me :+1:

**Comment by andrewvaughan** [comment]
I've updated pull request #2305 to reflect this
